### ISSUE TITLE
[Linux] Joystick button mapping not saved with non-English locale

### ISSUE THREAD

**Issue by sboukortt** [visualboyadvance-m/visualboyadvance-m#126]
### How to reproduce

1. Ensure that, say, `fr_FR.UTF-8` is uncommented in `/etc/locale.gen` (and run `sudo locale-gen` if needed)

2. ```console
   env LC_ALL=fr_FR.UTF-8 visualboyadvance-m 
   ```

3. *Options* → *Entrée* → *Configurer…*

4. Configure Joystick mappings

5. *Valider*

6. Open the config dialog again: *Options* → *Entrée* → *Configurer…* (or, alternatively, look at `~/.vbam/vbam.conf`)

### Expected behavior

Previously-configured joystick mappings appear and can be used in game.

### Actual behavior

Button mappings are missing, unless their English and French names coincide (e.g. `Joy1-HatN`, `Joy1-HatS`, `Joy1-HatE`).

**Note:** mappings that were saved when using VBA-M in English are *loaded* properly in French (both in game and in the configuration dialog), but they are lost if you try to modify them.

### Build details

Operating System: Arch Linux

If using Linux, specify if using xorg or Wayland: xorg

Version of code: 997b97573f30948d1dd7fc7f214ca3ebb0d85c7f

### Hardware details:

Joystick: Nacon GC-100XF

----

The problem seems to be that joystick button names are localized, e.g. `Joy1-Axis1-` becomes `Joy1-Axe1-` and `Joy1-Button0` becomes `Joy1-Bouton0`.

**Comment by ZachBacon** [comment]
This is a duplicate of #110

**Comment by sboukortt** [comment]
Oh, my bad. I missed it when looking at open issues.

**Comment by ZachBacon** [comment]
No worries, but it is an issue we are working on fixing. I shall keep this open though.

**Comment by adrienverge** [comment]
For those looking for a temporary solution to play with their joystick (these instructions are for Fedora / Gnome but should work in other environments):

1. Edit `/usr/share/applications/wxvbam.desktop` and replace the `Exec` line with:
   ```
   Exec=bash -c "LANG=en_US visualboyadvance-m %f"
   ```
2. Open VBA-M and configure your joystick in *Input* → *Configure*...
3. Close.
4. Restore the original contents of `/usr/share/applications/wxvbam.desktop`, in my case:
   ```
   Exec=visualboyadvance-m %f
   ```
5. Open VBA-M again: joystick commands should work. However DON'T GO TO *Input* → *Configure* AGAIN!

**Comment by rkitover** [comment]
@adrienverge you can also run the program from the terminal with this env var, no need to edit anything.

Thank you for the tip!

**Comment by v1993** [comment]
Easy way is to just run

    LANG=C visualboyadvance-m
in terminal, make all required changes in inputs (Input → Configure...), then close VBA-M and open it normal way. Easier, isn't it?

**Comment by denisfa** [comment]
Fixed with #437. We use english for all input, so this issue is no more.
